src/index.ts

```diff
@@ -0,0 +1,104 @@
+import express from 'express';
+import { readdir } from 'fs/promises'
+import path from 'path';
+import helmet from 'helmet';
+import { cwd } from 'process';
+import { extname, join } from 'upath'
+
+import 'dotenv/config'
+
+
+let app = express();
+const PORT = 3000;
+
+app.use(helmet())
+app.use(express.json())
+
+
+async function getDirectories(directory_path: string) {
+    const directories = (await readdir(path.resolve(directory_path), {
+        withFileTypes: true
+    }))
+        .filter(dirent => dirent.isDirectory())
+        .map(dirent => dirent.name);
+
+    return directories;
+}
+
+async function getFiles(directory_path: string) {
+    const files = (await readdir(path.resolve(directory_path), {
+        withFileTypes: true
+    }))
+        .filter(dirent => dirent.isFile())
+        .map(dirent => dirent.name);
+
+    return files;
+}
+
+function getDirectoryPath(path: string) {
+    return process.env.ROOT_DIR === path ? process.env.ROOT_DIR : cwd();
+}
+
+//change to correct base directory
+const rootDirectory = '/c/User/david/workspace///';
+
+app.get('/', (_, res) => {
```

Comment:
Good idea to list the endpoints. Let's use `/api` instead of `/` though.
```suggestion
app.get('/api', (_, res) => {
```